docs/graph.md

```diff
@@ -0,0 +1,714 @@
+# Graphs
+
+!!! danger "Don't use a nail gun unless you need a nail gun"
+    If PydanticAI [agents](agents.md) are a hammer, and [multi-agent workflows](multi-agent-applications.md) are a sledgehammer, then graphs are a nail gun, with flames down the side:
+
+    * sure, nail guns look cooler than hammers
+    * but nail guns take a lot more setup than hammers
+    * and nail guns don't make you a better builder, they make you a builder with a nail gun
+    * Lastly, (and at the risk of torturing this metaphor), if you're a fan of medieval tools like mallets and untyped Python, you probably won't like nail guns or our approach to graphs. (But then again, if you're not a fan of type hints in Python, you've probably already bounced off PydanticAI to use one of the toy agent frameworks — good luck, and feel free to borrow my sledgehammer when you realize you need it)
+
+    In short, graphs are a powerful tool, but they're not the right tool for every job. Please consider other [multi-agent approaches](multi-agent-applications.md) before proceeding.
+
+    Unless you're sure you need a graph, you probably don't.
```

Comment:
```suggestion
    If you're not confident a graph-based approach is a good idea, it's probably not necessary.
```
Just trying to soften the language a bit, the current phrasing somehow sounds just lightly condescending to me, if not toward the reader, then toward the library.